wa/workloads/motionmark/uiauto/app/src/main/java/com/arm/wa/uiauto/motionmark/UiAutomation.java

```diff
@@ -0,0 +1,182 @@
+/*    Copyright 2014-2018 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.arm.wa.uiauto.motionmark;
+
+import android.os.Bundle;
+import android.support.test.runner.AndroidJUnit4;
+import android.support.test.uiautomator.UiObject;
+import android.support.test.uiautomator.UiObjectNotFoundException;
+import android.support.test.uiautomator.UiSelector;
+import android.support.test.uiautomator.UiScrollable;
+
+import com.arm.wa.uiauto.BaseUiAutomation;
+import android.util.Log;
+
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.util.concurrent.TimeUnit;
+
+@RunWith(AndroidJUnit4.class)
+public class UiAutomation extends BaseUiAutomation {
+
+    private int networkTimeoutSecs = 30;
+    private long networkTimeout =  TimeUnit.SECONDS.toMillis(networkTimeoutSecs);
+    public static String TAG = "UXPERF";
+    public boolean textenabled = false;
+
+    @Before
+    public void initialize(){
+        initialize_instrumentation();
+    }
+
+    @Test
+    public void setup() throws Exception{
+        setScreenOrientation(ScreenOrientation.RIGHT);
+        clearFirstRun();
+    }
+
+    @Test
+    public void runWorkload() throws Exception {
+        runBenchmark();
+    }
+
+    @Test
+    public void teardown() throws Exception{
+        clearTabs();
+        unsetScreenOrientation();
+    }
+
+    public void clearFirstRun() throws Exception {
+        UiObject accept =
+            mDevice.findObject(new UiSelector().resourceId("com.android.chrome:id/terms_accept")
+                .className("android.widget.Button"));
+        if (accept.exists()){
+            accept.click();
+            UiObject negative =
+                mDevice.findObject(new UiSelector().resourceId("com.android.chrome:id/negative_button")
+                    .className("android.widget.Button"));
+            negative.waitForExists(100000);
+            negative.click();
+        }
+    }
+
+    public void runBenchmark() throws Exception {
+        setScreenOrientation(ScreenOrientation.RIGHT);
+        UiScrollable list = new UiScrollable(new UiSelector().scrollable(true));
+
+        UiObject start =
+            mDevice.findObject(new UiSelector().text("Run Benchmark")
+                .className("android.widget.Button"));
+        list.scrollIntoView(start);
+        start.click();
+
+        setScreenOrientation(ScreenOrientation.NATURAL);
```

Comment:
This is rotating back once the test has started. 

The workflow should be 

Portrait by default when launched
Switch to landscape to run the tests
Return to portrait to see all the results on one screen

Reply:
Ok, in which case should that call not be after `results.waitForExists(2100000);` as this will signify the tests have completed and the results are ready to be extracted?

Reply:
Yep, which I changed to on the latest commit after revisiting :)

Reply:
Well I thought I had but it's not showing here so I will redo that.